framework/src/mesh/MooseMesh.C

```diff
@@ -3181,17 +3181,43 @@ MooseMesh::buildFaceInfo() const
   _all_face_info.clear();
   _elem_side_to_face_info.clear();
 
+  _elem_to_elem_info.clear();
+  _internal_elem_info.clear();
+  _elem_to_ghost_info.clear();
+
   // by performing the element ID comparison check in the below loop, we are ensuring that we never
   // double count face contributions. If a face lies along a process boundary, the only process that
   // will contribute to both sides of the face residuals/Jacobians will be the process that owns the
   // element with the lower ID.
   auto begin = getMesh().active_elements_begin();
   auto end = getMesh().active_elements_end();
 
-  for (auto it = begin; it != end; ++it)
+  // We use a counter, to speed up this process (the same ordering will be used
+  // in the loop below). For users who want to access the ElemInfo based on
+  // Elem*, we prepare a map connecting the Elem* and the index in the ElemInfo
+  // vector.
```

Comment:
Does this really speed things up? I just wonder if we need two containers?

Reply:
Well, my assumption was that storing the eleminfos in a contagious-in-memory storage is more efficient. In terms of access, we would have to do some sort of lookup and I thought storing unsigned ints instead of ElemInfo* might save some memory (4B instead of 8B, I think). Now that I started thinking about it, I realize that this counter should be a dof_id_type in cases we have a lot of cells. So in many cases we will have 8B for that too. Anyways, The faceinfo is handled similarly, there is a vector-storage and a map for lookup.

Reply:
I am trying to learn these CS-related things so I am not super confident if this is actually faster. I might modify the comment if you think the current approach is acceptable, if not I'll just modify the container.

Reply:
Sequential container lookup is definitely faster. It's constant complexity whereas a map is O(log N) and `unordered_map` lookup is amortized constant but you also have to pay the cost of hashing the key. But when I look at the code I see that one of your sequential container lookups requires you to query the map anyway to get the `dof_id_type` for the neighbor. So if you already have to do the map lookup anyway for one container sequencing, I think that leaves only one other place in the code where you do the container lookup. Unless that lookup really saves a lot it seems like you could save the internal `ElemInfo` instances in the map container and drop the sequential container. Then you could also remove the `counter`. It's a fair point that we have both containers for `FaceInfo` but we also have uses outside of `MooseMesh` for the face info vector/sequential container

Reply:
Got it, I'll change it with the next commit!

Reply:
Okay, I read a bit about containers and I changed the implementation. Also, do you think it would be possible to do this based on elem_ids? Then we could keep the vector container alone. But I am not 100% sure how this behaves with local refinement.

Reply:
The problem with that is that element IDs are not guaranteed to be continguous so you could have gaps in the vector, which would be problematic since you'd have to default construct `ElemInfo` and use unnecessary memory for those gaps.